quiz/quiz.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bufio"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+	"unicode"
+)
+
+type problems struct {
+	q string
+	a string
+}
+
+func main() {
+	const defaultName = "problems.csv"
+	const defaultTimeLimit = 30
+	fileName := flag.String("csv", defaultName, "csv file in the format: `question,answer`")
+	timeLimit := flag.Int("limit", defaultTimeLimit, "Integer time limit ")
+	flag.Parse()
+
+	if isValidFileName(*fileName) {
+		arr, err := readFile(*fileName)
+		if err != nil {
+			log.Fatal(err)
+			return
+		}
+		problems := parseQuiz(arr)
+		quiz(problems, *timeLimit)
+	} else {
+		log.Fatal("Invalid file name")
+		return
+	}
+}
+
+func quiz(problems []problems, timeLimit int) {
+	correct := 0
+	incorrect := 0
+	timer := time.NewTimer(time.Duration(timeLimit) * time.Second)
+
+	for idx, prob := range problems {
+		fmt.Printf("Question number %d is %s? \n", idx+1, prob.q)
+		select {
+		case <-timer.C:
+			fmt.Println("Ops! You ran out of time")
```

Comment:
I waited for more than 30 seconds and it never got to this part of the code

Reply:
Yikes! I wonder what happened there. I'll check

Reply:
Did you answer all the questions before 30 seconds? If you did, that explains it. The CSV file has only 5 questions so I usually pass in a 5 second limit when running the code

Reply:
I did not answer all the questions before 30 seconds. I answered like two and waited to see if it would get to that timer.

Reply:
So I think the bug is, you need to answer before it throws the error. I'll fix that